BraintreeCore/src/main/java/com/braintreepayments/api/core/ErrorWithResponse.kt

```diff
@@ -98,18 +105,19 @@ data class ErrorWithResponse internal constructor(
             try {
                 val errors = json
                     ?.let { JSONObject(it) }
-                    ?.getJSONArray(GraphQLConstants.Keys.ERRORS)
+                    ?.optJSONArray(GraphQLConstants.Keys.ERRORS)
+                    ?: throw JSONException("Errors key not found in JSON")
                 errorWithResponse.fieldErrors = BraintreeError.fromGraphQLJsonArray(errors)
 
                 val fieldErrorsEmpty = errorWithResponse.fieldErrors?.isEmpty() ?: true
                 errorWithResponse.message = if (fieldErrorsEmpty) {
-                    errors?.getJSONObject(0)?.getString(GraphQLConstants.Keys.MESSAGE)
+                    errors.getJSONObject(0)?.getString(GraphQLConstants.Keys.MESSAGE)
                 } else {
                     ErrorMessages.USER
                 }
             } catch (e: JSONException) {
                 errorWithResponse.apply {
-                    message = "Parsing error response failed"
```

Comment:
Does this message no longer need to be set?

Reply:
Yes, I think so because USER like to know more specific error messages.
Also, i think we should 
->Check API Responses to ensure that the API or server responses are in the expected format and properly structured and those keys are matching.
->Improve Error Handling: Add better error handling to provide more specific error messages instead of generic ones like "Parsing error response failed."
->Log Errors for Debugging: Use logging to capture the exact issue when parsing fails to help diagnose and fix the problem from development side.